External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -553,14 +550,7 @@ namespace FEXCore::Context {
   }
 
   FEXCore::Core::InternalThreadState* Context::CreateThread(FEXCore::Core::CPUState *NewThreadState, uint64_t ParentTID) {
-    FEXCore::Core::InternalThreadState *Thread{};
-
-    // Grab the new thread object
-    {
-      std::lock_guard<std::mutex> lk(ThreadCreationMutex);
-      Thread = Threads.emplace_back(new FEXCore::Core::InternalThreadState{});
-      Thread->ThreadManager.TID = ++ThreadID;
```

Comment:
Why is there no equivalent to this line in the new code?

Reply:
This TID is incorrect, and seems like some leftover from when FEX implementing its own TID and PIDs. `ThreadID` isn't even an atomic.

The correct value is set by `Context::InitializeThreadTLSData` called from `Context::ExecutionThread`, which is run with the host backing thread created from `Context::InitializeThread`.

Reply:
(The actual thread creation is done using a custom threading implementaiton in `FEXCore/Source/Utils/Thread.cpp `class PThread`)